src/ir/module.h

```diff
@@ -34,11 +34,13 @@ class Block;
 class Operation {
  public:
   Operation(const Block* parent, const Operator& op,
-            NamedAttributeMap attributes, NamedValueMap inputs)
+            NamedAttributeMap attributes, NamedValueMap inputs,
+            std::unique_ptr<Module> module = nullptr)
```

Comment:
Would it make for cleaner calls to pass `module` in as type `Module` and have `Operation` put it in a `std::unique_ptr`... or even just embed it in a `std::optional<Module>`?

Reply:
This is similar to your question about `AddBlock(Block block)`. Let us discuss there?